### ISSUE TITLE
Conversion to tensorrt failed at some nms nodes

### ISSUE THREAD

**Issue by yeungkong** [open-mmlab/mmdeploy#953]
Hi developers. When I try to convert my mask rcnn model to tensorrt, the conversion failed at some nms node. In my model, the nms_pre and max_per_img parameters are set to 8000. When I try to do the conversion, the following error message occurs.

```
....
WARNING: The shape inference of mmdeploy::grid_sampler type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
2022-08-27 09:24:01,609 - mmdeploy - INFO - Execute onnx optimize passes.
2022-08-27 09:24:02,119 - mmdeploy - INFO - Finish pipeline mmdeploy.apis.pytorch2onnx.torch2onnx
2022-08-27 09:24:04,575 - mmdeploy - INFO - Start pipeline mmdeploy.backend.tensorrt.onnx2tensorrt.onnx2tensorrt in subprocess
2022-08-27 09:24:04,775 - mmdeploy - WARNING - Could not load the library of tensorrt plugins.             Because the file does not exist:
[08/27/2022-09:24:05] [TRT] [I] [MemUsageChange] Init CUDA: CPU +350, GPU +0, now: CPU 9495, GPU 2423 (MiB)
[08/27/2022-09:24:06] [TRT] [I] [MemUsageChange] Init builder kernel library: CPU +339, GPU +104, now: CPU 10013, GPU 2527 (MiB)
[08/27/2022-09:24:06] [TRT] [W] onnx2trt_utils.cpp:369: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[08/27/2022-09:24:06] [TRT] [W] onnx2trt_utils.cpp:395: One or more weights outside the range of INT32 was clamped
[08/27/2022-09:24:06] [TRT] [W] onnx2trt_utils.cpp:395: One or more weights outside the range of INT32 was clamped
[08/27/2022-09:24:08] [TRT] [E] [layers.cpp::nvinfer1::TopKLayer::TopKLayer::4274] Error Code 3: API Usage Error (Parameter check failed at: layers.cpp::nvinfer1::TopKLayer::TopKLayer::4274, condition: k > 0 && k <= MAX_TOPK_K
)
Process Process-3:
Traceback (most recent call last):
  File "D:\Program_Files\Anaconda3\envs\..\lib\multiprocessing\process.py", line 297, in _bootstrap
    self.run()
  File "D:\Program_Files\Anaconda3\envs\..\lib\multiprocessing\process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "..\mmdeploy\mmdeploy\apis\core\pipeline_manager.py", line 107, in __call__
    ret = func(*args, **kwargs)
  File "..\mmdeploy\mmdeploy\backend\tensorrt\onnx2tensorrt.py", line 88, in onnx2tensorrt
    device_id=device_id)
  File "..\mmdeploy\mmdeploy\backend\tensorrt\utils.py", line 113, in from_onnx
    raise RuntimeError(f'Failed to parse onnx, {error_msgs}')
RuntimeError: Failed to parse onnx, In node 1013 (importTopK): UNSUPPORTED_NODE: Assertion failed: layer && "Failed to add TopK layer."

2022-08-27 09:24:09,067 - mmdeploy - ERROR - `mmdeploy.backend.tensorrt.onnx2tensorrt.onnx2tensorrt` with Call id: 1 failed. exit.
```

So I try to reduce the nms_pre and max_per_img parameters to 3000. Then the following error message occurs.

```
...
WARNING: The shape inference of mmdeploy::grid_sampler type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
2022-08-27 09:32:26,844 - mmdeploy - INFO - Execute onnx optimize passes.
2022-08-27 09:32:27,374 - mmdeploy - INFO - Finish pipeline mmdeploy.apis.pytorch2onnx.torch2onnx
2022-08-27 09:32:29,844 - mmdeploy - INFO - Start pipeline mmdeploy.backend.tensorrt.onnx2tensorrt.onnx2tensorrt in subprocess
2022-08-27 09:32:30,052 - mmdeploy - WARNING - Could not load the library of tensorrt plugins.             Because the file does not exist:
[08/27/2022-09:32:30] [TRT] [I] [MemUsageChange] Init CUDA: CPU +357, GPU +0, now: CPU 9455, GPU 2423 (MiB)
[08/27/2022-09:32:31] [TRT] [I] [MemUsageChange] Init builder kernel library: CPU +338, GPU +104, now: CPU 9984, GPU 2527 (MiB)
[08/27/2022-09:32:31] [TRT] [W] onnx2trt_utils.cpp:369: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[08/27/2022-09:32:31] [TRT] [W] onnx2trt_utils.cpp:395: One or more weights outside the range of INT32 was clamped
[08/27/2022-09:32:31] [TRT] [W] onnx2trt_utils.cpp:395: One or more weights outside the range of INT32 was clamped
[08/27/2022-09:32:35] [TRT] [I] No importer registered for op: TRTBatchedNMS. Attempting to import as plugin.
[08/27/2022-09:32:35] [TRT] [I] Searching for plugin: TRTBatchedNMS, plugin_version: 1, plugin_namespace:
Process Process-3:
Traceback (most recent call last):
  File "D:\Program_Files\Anaconda3\envs\..\lib\multiprocessing\process.py", line 297, in _bootstrap
    self.run()
  File "D:\Program_Files\Anaconda3\envs\..\lib\multiprocessing\process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "..\mmdeploy\mmdeploy\apis\core\pipeline_manager.py", line 107, in __call__
    ret = func(*args, **kwargs)
  File "..\mmdeploy\mmdeploy\backend\tensorrt\onnx2tensorrt.py", line 88, in onnx2tensorrt
    device_id=device_id)
  File "..\mmdeploy\mmdeploy\backend\tensorrt\utils.py", line 113, in from_onnx
    raise RuntimeError(f'Failed to parse onnx, {error_msgs}')
RuntimeError: Failed to parse onnx, In node 1237 (importFallbackPluginImporter): UNSUPPORTED_NODE: Assertion failed: creator && "Plugin not found, are the plugin name, version, and namespace correct?"

2022-08-27 09:32:35,526 - mmdeploy - ERROR - `mmdeploy.backend.tensorrt.onnx2tensorrt.onnx2tensorrt` with Call id: 1 failed. exit.
```

I need the parameters to be 8000 and I need the conversion to be successful. How can I fix the problems? I really need your help. Thank you.

**Comment by hanrui1sensetime** [comment]
Hi, @yeungkong , what about the conversion result if you use the original topk parameters instead of 8000? Successful or failed?

**Comment by yeungkong** [comment]
@hanrui1sensetime I reinstalled cuda, cudnn, mmdetection and mmdeploy. In particular, I reinstalled tensorrt, onnxruntime and mmdeploy more carefully strictly according to https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/01-how-to-build/windows.md. For onnxruntime, I installed the windows prebuilt binary package from https://github.com/microsoft/onnxruntime/releases/tag/v1.8.1, and I used the command `pip install onnxruntime` to install onnxruntime1.12.1 afterwards. I think now the error at nms nodes disappears but now some illegal memory access(https://github.com/open-mmlab/mmdeploy/issues/965) occurs .
